### PR TITLE
less clicks on save-message and start-chat

### PR DESCRIPTION
**Pull request by r10s**
this pr streamlines the ui, by removing some superfluous alerts and by re-ordering the forward-dialog so that "saved messages" is always one of the first entries.

counterpart of https://github.com/deltachat/deltachat-android/pull/1266 and https://github.com/deltachat/deltachat-core-rust/pull/1336/ , the core-part needs to be merged first

### CODE REVIEW

**Comment by cyBerta** [comment]
@r10s due to other merged PRs this one has a conflict now

**Comment by r10s** [comment]
@cyBerta i resolved all conflicts

**Comment by cyBerta** [comment]
UI-wise I think having the saved messages below the pinned chats would be nicer.
Besides that it's a really nice feature.

**Comment by r10s** [comment]
> UI-wise I think having the saved messages below the pinned chats would be nicer.
Besides that it's a really nice feature.

i fact, this was my first approach. but, after using it some days this way, i found out that it is better to have the "saved messages" _always_ at the same, first position.
also as this lacks the confirmation-dialog, it is maybe a bit easier to understand that the first item is a bit different.

File: deltachat-ios/Controller/ChatListController.swift [hunk 1]
```diff
@@ -157,8 +161,9 @@ extension ChatListController: UITableViewDataSource, UITableViewDelegate {
         }
 
         let chatId = chatList.getChatId(index: row)
+        let chat = DcChat(id: chatId)
         if chatId == DC_CHAT_ID_ARCHIVED_LINK {
-            return getArchiveCell(tableView)
+            return getArchiveCell(tableView, title: chat.name)
```
Comment (cyBerta) [inline deltachat-ios/Controller/ChatListController.swift]: wow, the chat name has now the translated string for "Archived chats (#noOfChats)"?

Comment (r10s) [reply deltachat-ios/Controller/ChatListController.swift]: yip, exactly :)

**Pull request merged**